src/java/com/zimbra/cs/taglib/tag/LoginTag.java

```diff
@@ -151,9 +164,19 @@ public void doTag() throws JspException, IOException {
                 options.setAuthToken(mAuthToken);
                 options.setAuthAuthToken(true);
             } else {
+                String virtualHost = getVirtualHost(request);
+
+                if (mUsername != null && !mUsername.isEmpty() && mUsername.indexOf("@") != -1) {
+                    String usernameSplit[]= mUsername.split("@");
+
+                    // check if user domain matches current virtual host.
+                    if (!virtualHost.equals(usernameSplit[1])) {
+                        throw AuthFailedServiceException.AUTH_FAILED(mUsername, "", "Invalid username for virtual host = ".concat(virtualHost));
```

Comment:
nit: remove word "virtual" from exception error?  I believe it would be preferable to leave the hostname out of the exception completely.  Is that info not already available on the server logs?  Or conversely, do we also need the username logged with this request (if it is not already logged?).